python/ray/_private/log.py

```diff
@@ -89,3 +92,21 @@ def generate_logging_config():
         # See https://github.com/ray-project/ray/pull/31858 for related PR
         rllib_logger = logging.getLogger("ray.rllib")
         rllib_logger.setLevel(logging.WARN)
+
+        # Set up the LogRecord factory.
```

Comment:
Separate the logic into a function so that it is easy to unit test, and we can write tests such as `class TestSetupLogRecordFactory`.